src/core/lv2/Lv2Proc.cpp

```diff
@@ -447,6 +689,27 @@ void Lv2Proc::createPorts()
 				else if (!portRef->m_right) { portRef->m_right = &audio; }
 			}
 		}
+
+		void visit(Lv2Ports::AtomSeq& atomPort) override
+		{
+			if(atomPort.m_flow == Lv2Ports::Flow::Input)
+			{
+				if(atomPort.flags & Lv2Ports::AtomSeq::FlagType::Midi)
+				{
+					Q_ASSERT(!m_proc->m_midiIn);
+					m_proc->m_midiIn = &atomPort;
```

Comment:
Is it worth checking in `Lv2Proc::check` that at most one MIDI in (and out) port is present? Otherwise plugins with multiple MIDI in/out ports can appear in the instrument list, and LMMS will exit with an assertion failure if the user tries to add one to a project.

Reply:
Good point. I fixed that now.